phpseclib/Net/SSH2.php

```diff
@@ -2132,9 +2144,11 @@ function _keyboard_interactive_process()
      */
     function _ssh_agent_login($username, $agent)
     {
+        $this->agent = $agent;
         $keys = $agent->requestIdentities();
         foreach ($keys as $key) {
             if ($this->_privatekey_login($username, $key)) {
+                $this->agent->_on_login_success($this);
```

Comment:
> It only does this if you request it (i.e.) from startSSHForwarding

I see. Overall, I think this is a rather ugly solution, but let's deal with https://github.com/phpseclib/phpseclib/pull/592#issuecomment-69684541 first.